mixer/pkg/runtime/dispatcher/tracing.go

```diff
@@ -31,9 +32,19 @@ const (
 	errorStr     = "error"
 )
 
-// LogToDispatchSpan logs to the given Span in a structured manner. Span must be valid.
-func logToDispatchSpan(span opentracing.Span, template string, handler string, adapter string, err error) {
-	st := status.OK
+func logCheckResultToDispatchSpan(span opentracing.Span, template string, handler string, adapter string, result adapter.CheckResult, err error) {
+	logEntriesToDispatchSpan(span, template, handler, adapter, result.Status, err)
+}
+
+func logQuotaResultToDispatchSpan(span opentracing.Span, template string, handler string, adapter string, result adapter.QuotaResult, err error) {
+	logEntriesToDispatchSpan(span, template, handler, adapter, result.Status, err)
+}
+
+func logErrorToDispatchSpan(span opentracing.Span, template string, handler string, adapter string, err error) {
+	logEntriesToDispatchSpan(span, template, handler, adapter, status.OK, err)
+}
+
+func logEntriesToDispatchSpan(span opentracing.Span, template string, handler string, adapter string, st rpc.Status, err error) {
```

Comment:
@robertpanzer This seems like a good place to short circuit the call is the current span is not being sampled.
`span.LogFields` is going to assemble arguments in a slice. Considering 1% trace sampling rate, it would do this needless work 99% of the time.

If it is not possible to get sampled status of this span, then I am ok with what you have.

Reply:
@mandarjog The flag that contains the information if the Span is sampled or not doesn't seem to be available from an opentracing SpanContext. 
Also the jaeger SpanContext doesn't expose this flag.